File: vending_machine.js
'use strict';
const prompt = require("prompt-sync")();

// Values for coin inputs
const coinValues = {
    q: 25,
    d: 10,
    n: 5
};

// List of products, containing names and costs
const productList = [
    {name: "Doritos", cost: 100},
    {name: "Takis", cost: 125},
    {name: "Mike & Ikes", cost: 65}
];

// Gather input to build up total amount of money
let done = false;
while (!done) {
    let coin = prompt("Enter a coin: Q, N, D, or S to select item: ");
    if (coin == 's') {
        done = true;
    }
    else if (coin in coinValues) {
        total += coinValues[coin];
        console.log(`Total is now: ${toMoney(total)}`);
    }
    else {
        console.log("I don't know what that is.");
    }
}

while (total < minimum) {
    printChoices(productList);
}


// Function to format a number of cents into a readable dollar amount
function toMoney(value) {
    return "$"+(value/100).toFixed(2);
};

// Prints the choices to the user from the product list
function printChoices(products) {
    for (let i = 0; i < products.length; i++) {
        console.log(` ${i+1}. ${products[i].name}: ${products[i].cost}`);
    }
}

// Find the minimum cost to buy something
function findMinimumCost(products) {
    let minimum = 1000;
    for (let i = 0; i < products.length; i++) {
        if (products[i].cost < minimum) {
            minimum = products[i].cost;
        }
    }
    return minimum;
}

File: vending_machine.py
# from decimal import *

# getcontext().prec = 2
# total = Decimal(0);

#  This is the start of the program

total = 0
while 1:
    print ("You've inserted: "+"${:,.2f}".format(total/100))
    coin = input("> Enter coin (Q, D, N, X, DC): ").lower()
    if coin == 'x':
        break
    elif coin == 'q':
        total += 25
    elif coin == 'd':
        total += 10
    elif coin == 'n':
        total += 5
    elif coin == 'dc':
        total += 100
    else:
        print ("Unknown coin type")
    
while (total >= 65):

    print ("\nYou have "+"${:,.2f}".format(total/100)+" to spend.")
    print ("What would you like?")
    print (" 1. Doritos - $.75")
    print (" 2. Takis - $1.00")
    print (" 3. Mike & Ikes - $0.65")

    cost = 0
    while (cost == 0):
        choice = input("> Choice: ")
        if choice == '1':
            cost = 75
        elif choice == '2':
            cost = 100
        elif choice == '3':
            cost = 65
        else:
            print ("Invalid choice. Try again.")

    change = total - cost
    if change >= 0:
        print ("Your change: "+"${:,.2f}".format(change/100))
    else:
        print ("Sorry, you're "+"${:,.2f}".format(abs(change)/100)+" short.")
    total -= cost

